src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,109 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderRequestBody } from '../../../order';
+import Payment from '../../payment';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { GooglePayPaymentStrategy, PaymentMethodData } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomGooglePayPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._googlePayOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+
+        const testMode: boolean = !!getPaymentMethodOrThrow(methodId).config.testMode;
+        const { checkoutcomkey } = getPaymentMethodOrThrow(methodId).initializationData;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
```

Comment:
@davidchin i have a question about this one, currently execute is override and doesn't have async, so:
Currently:
    On throw: throw exception
If changed to async execute:
    On throw: return Rejected promise

I'm not sure if this will have an impact on top level code